spirv_glsl.cpp

```diff
@@ -16856,7 +16856,12 @@ void CompilerGLSL::emit_function(SPIRFunction &func, const Bitset &return_flags)
 			{
 				// Recursively emit functions which are called.
 				uint32_t id = ops[2];
-				emit_function(get<SPIRFunction>(id), ir.meta[ops[1]].decoration.decoration_flags);
+				auto& calledFunction = get<SPIRFunction>(id);
+
+				// Propagate state back up the call tree.
+				func.emits_geometry |= calledFunction.emits_geometry;
```

Comment:
While convenient, the parsed data structures should be isolated from the per-backend analysis. This information can be propagated during that analysis.